unit/src/RankFourTensorTest.C

```diff
@@ -28,6 +28,32 @@ TEST(RankFourTensor, invSymm1)
   EXPECT_NEAR(0, (iSymmetric - a.invSymm() * a).L2norm(), 1E-5);
 }
 
+TEST(RankFourTensor, symmetric21)
+{
+  // inverse check for full anisotropic tensor
+  std::vector<Real> input(21);
+  for (size_t i = 0; i < input.size(); i++)
+  {
+    input[i] = i + 1;
+  }
+
+  RankFourTensor a(input, RankFourTensor::symmetric21);
+  EXPECT_NEAR(0, (iSymmetric - a.invSymm() * a).L2norm(), 1E-5);
```

Comment:
No! This is not a test for the fill method at all. As discussed in the call you need to have a gold standard tensor (with _all_ 81 entries). Prepare that using the unmodified fill method. Then you compare if the new fill method produced the same result. 

Your test would pass for _any_ invertible tensor! It just tests `invSym` again. 

Same for the function below.